### PR TITLE
Use builder directly for OpAll/OpAny

### PR DESCRIPTION
**Pull request by trenouf**
This commit changes the SPIR-V reader to generate IR directly for OpAll
and OpAny.

Change-Id: I3e63acb3645894bdb9f771876c815edf88fcae35

### CODE REVIEW

**Comment by trenouf** [comment]
Please review just the topmost commit

File: translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -5990,9 +5989,24 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
...
-                    transOCLAllAny(static_cast<SPIRVInstruction *>(BV), BB));
+  case OpAny: {
+    SPIRVUnary *BC = static_cast<SPIRVUnary *>(BV);
+    Value *Val = transValue(BC->getOperand(0), F, BB);
+    if (!isa<VectorType>(Val->getType()))
+      return Val;
+    Value *Result = getBuilder()->CreateExtractElement(Val, uint64_t(0));
+    for (unsigned I = 1, E = Val->getType()->getVectorNumElements(); I != E;
+         ++I) {
+      Value *El = getBuilder()->CreateExtractElement(Val, I);
```
Comment (amdrexu) [inline translator/lib/SPIRV/SPIRVReader.cpp]: El ->Elem

**Comment by trenouf** [comment]
> Hi, tim,
> 
> Is it necessary to add m_pBuilder->SetInsertPoint(BB) here?

No, because it now does SetInsertPoint before the big switch statement.

**Comment by trenouf** [comment]
V2: Addressed review comments.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5c9c012734dc0de6016b5d04891e4c3ecf8f9185
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 88d997710b4e405f3a8e3fd60a38afee9e3e77e2</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 2cb5558b94c5dc839e093cb439057a1802426c8e</li>
<li>LLVM: amd-vulkan-dev dcb42575c3d374467bb6e4e609f8bf6d1302c297</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 927/927 (100.0%)</li>
<li>Failed: 0/927 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 68331/68331 (100.0%)</li>
<li>Failed: 0/68331 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 68363/68363 (100.0%)</li>
<li>Failed: 0/68363 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**